R/build_model.R

```diff
@@ -41,43 +41,47 @@ build_model_ <- function(data, model_func, seed = 0, test_rate = 0, group_cols =
     stop("test_rate has to be between 0 and 1")
   }
 
+  # use this for preprocess
+  processed <- data
   # avoid name conflict of grouping columns
-  colnames(data)[group_col_index] <- avoid_conflict(
+  colnames(processed)[group_col_index] <- avoid_conflict(
     reserved_names,
-    colnames(data)[group_col_index],
+    colnames(processed)[group_col_index],
     ".group"
   )
 
   # make column names unique
-  colnames(data) <- make.unique(colnames(data), sep = "")
+  colnames(processed) <- make.unique(colnames(processed), sep = "")
 
   if(!is.null(group_cols)){
-    data <- dplyr::group_by_(data, .dots = colnames(data)[group_col_index])
-  } else if (!dplyr::is.grouped_df(data)){
+    processed <- dplyr::group_by_(processed, .dots = colnames(processed)[group_col_index])
+  } else if (!dplyr::is.grouped_df(processed)){
     # need to be grouped to nest
-    data <- data %>%
+    processed <- processed %>%
       dplyr::mutate(.test_index = 1) %>%
       dplyr::group_by(.test_index)
   }
 
-  group_col_names <- grouped_by(data)
+  group_col_names <- grouped_by(processed)
 
   # integrate all additional parameters
   dots <- lazyeval::all_dots(.dots, ...)
 
   # check if variables in grouped_col_names are not used
   formula <- dots$formula
   if(!is.null(formula)){
-    vars <- all.vars(formula$expr)
+    # expand formula with column names of data
```

Comment:
Please explain that in a comment so that we accumulate the knowledge of why we do it certain way.